l1-contracts/contracts/upgrades/BaseZkSyncUpgrade.sol

```diff
@@ -253,8 +253,14 @@ abstract contract BaseZkSyncUpgrade is ZKChainBase {
             return bytes32(0);
         }
 
-        if (_l2ProtocolUpgradeTx.txType != SYSTEM_UPGRADE_L2_TX_TYPE) {
-            revert InvalidTxType(_l2ProtocolUpgradeTx.txType);
+        if (s.boojumOS) {
```

Comment:
to reduce code repetition as well as chance that someone might accidentally forget to add such an if, could you please make an internal function in `ZKChainBase` called `_getPriorityTxType()` and `_getUpgradeTxType()` and it would compare against the `s.boojumOS` and return the corresponding constant? otherwise lgtm